src/swapd/runtime.rs

```diff
@@ -2105,6 +2144,66 @@ impl Runtime {
                 };
             }
 
+            Request::CancelSwap
+                if self.state.a_start() || self.state.a_commit() || self.state.a_reveal() =>
+            {
+                // just cancel the swap, no additional logic required
+                self.state_update(
+                    endpoints,
+                    State::Alice(AliceState::FinishA(Outcome::Cancel)),
+                )?;
+                self.cancel_swap(endpoints)?;
+                self.send_ctl(
+                    endpoints,
+                    source,
+                    Request::String("Canceled swap".to_string()),
+                )?;
+            }
+            Request::CancelSwap if self.state.b_start() => {
+                // just cancel the swap, no additional logic required, since funding was not yet retrieved
+                self.state_update(endpoints, State::Bob(BobState::FinishB(Outcome::Cancel)))?;
+                self.cancel_swap(endpoints)?;
+                self.send_ctl(
+                    endpoints,
+                    source,
+                    Request::String("Canceled swap".to_string()),
+                )?;
+            }
+            Request::CancelSwap
+                if self.state.b_commit()
+                    || self.state.b_reveal()
+                    || (!self.state.b_received_refund_procedure_signatures()
+                        && self.state.b_core_arb()) =>
```

Comment:
Wouldn't it be worth for readability to add a some btc funded field in those states?